src/components/ec/cuda/ec_cuda.c

```diff
@@ -403,7 +406,11 @@ ucc_status_t ucc_ec_cuda_event_create(void **event)
     ucc_ec_cuda_event_t *cuda_event;
 
     cuda_event = ucc_mpool_get(&ucc_ec_cuda.events);
-    ucc_assert(cuda_event);
+    if (ucc_unlikely(!cuda_event)) {
+        ec_error(&ucc_ec_cuda.super, "failed to get event from mpool");
+        return UCC_ERR_NO_MEMORY;
+    }
```

Comment:
Minor: perhaps we can add a macro later for this kind of mpool error since it would appear in multiple places.